### PR TITLE
sinkv2(cdc): add a memory table based sort engine for tests

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What problem does this PR solve?

Issue Number: ref #5928 

### What is changed and how it works?

Add a memory table based sort engine for tests

#### Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: pkg/sorter/sorter.go [hunk 1]
```diff
@@ -102,7 +102,7 @@ type Position struct {
 
 // Valid indicates whether the position is valid or not.
 func (p Position) Valid() bool {
-	return p.StartTs != 0 && p.CommitTs != 0
```
Comment (hi-rustin) [inline pkg/sorter/sorter.go:105]: Why do we remove start ts?

**Review by hi-rustin — APPROVED** [review]

File: pkg/sorter/sorter.go (hunk 1 above)
Comment (hicqu) [reply pkg/sorter/sorter.go:105]: In some tests transactions' `StartTs` can be 0, so just check `CommitTs` is OK.

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**